generate_Zscores/NormalizeExpressionLevels.py

```diff
@@ -0,0 +1,137 @@
+import sys
+import os
+import statistics
+import argparse
+
+# Given expression data for a set of samples generate normalized expression values (Independant of unaltered status from CNA file)
+# 
+# Each gene is normalized separately. First, the expression distribution of the gene is
+# estimated by calculating the mean and variance of the expression values for samples whose 
+# values are not Null, NA or NaN 
+#
+# If the gene has samples whose expression values are Null, NaN or 'NA', then its normalized expression is reported as NA. 
+# Otherwise, for every sample, the gene's normalized expression is reported as
+# 
+# (r - mu)/sigma
+#
+# where r is the raw expression value, and mu and sigma are the mean and standard deviation 
+# of the samples with expression values, respectively.
+#
+# The syntax is simple:
+# 
+# python NormalizeExpressionLevels.py -i <expression_file> -o <output_file>
+#
+# The output is written onto a file named "output_file"
+#
+# Any number of columns may precede the data. However, the following must be satisfied:
+#
+# - the first column provides gene identifiers
+#
+# Algorithm:
+# Input expression file
+# concept:
+# for each gene:
+# 		compute mean and standard deviation for samples ( n = # of samples where expression value is not Null, NA, NaN)
+#		for each sample:
+#		compute zScore when standard deviation != 0
+#
+
+
+# calculate mean and std for the samples whose values are not Null or NA (n)
+def calculate_mean_std(line,start_position):
+	data = line.split('\t')
+	data = data[start_position:]
+	# Remove strings '' or 'NA' from list 
+	for elem in data:
+		if elem == '' or elem == 'NA' or elem == 'NaN':
+			data.pop(data.index(elem))
+	# Convert string list to floats
+	for index, item in enumerate(data):
+		data[index] = float(item)
+	# Calculate mean and std
+	n = len(data)
+	mu = statistics.mean(data)
+	sigma = statistics.stdev(data,mu)
+	return(mu,sigma)
+
+# Calculate z_scores 	
+# TODO: what if standard deviation is 0.0?
```

Comment:
My understanding in this case, the program should exit and z-score can't be calculated for this case. @jjgao

Reply:
Yup. I thought the same and added the condition to skip the normalization of the row in line 130. I hope skipping the rows whose std is 0 is correct?

Reply:
Updated the script to output 'NA' for the genes whose standard deviation is zero instead of skipping the lines.